### PR TITLE
Add Warsow

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://discord.com/channels/592341480383971328/894361944990048276/974757979205283931
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)* https://github.com/Warsow/qfusion/pull/53

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by Mailaender** [comment]
bot, build net.warsow.Warsow

File: flathub.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: disappointing

File: warsow.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  <id>warsow.desktop</id>
+  <name>Warsow</name>
+  <summary>Fast paced 3D first person shooter</summary>
+  <description>
+    <p>Warsow is set in a futuristic cartoonish world where rocketlauncher-wielding pigs and lasergun-carrying cyberpunks roam the streets. It is a completely free, fast-paced first-person shooter for Windows, Linux, and Mac OS X.</p>
+    <p>Speed and movement is what Warsow is all about. Like a true cyberathlete you jump, dash, dodge, and walljump your way through the game. Grab power-ups before your enemy does, plant a bomb before anyone sees you, and steal the enemy’s flag before they know what is going on!</p>
+    <p>Our goal is to offer a fast and fun competitive first-person shooter without hard graphical violence - Warsow has no blood or guts flying around. Red stars instead of blood indicate hits and colored cubes replace guts as gib effects.</p>
+  </description>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-2.0+ AND CC-BY-SA-4.0 AND CC-BY-ND-4.0</project_license>
```
Comment (hfiguiere) [inline warsow.appdata.xml:12]: this shouldn't include the license of the assets.

File: net.warsow.Warsow.yaml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+          stable-only: true
+          url-template: https://github.com/ImageMagick/ImageMagick/archive/$version.tar.gz
+    cleanup:
+      - /bin
+      - /etc
+      - /lib/ImageMagick*
+      - /share/ImageMagick*
+
+  - name: warsow
+    buildsystem: cmake
```
Comment (hfiguiere) [inline net.warsow.Warsow.yaml:39]:
```suggestion
    buildsystem: cmake-ninja
```

unless it fails.

File: net.warsow.Warsow.yaml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      - type: archive
+        url: https://github.com/ImageMagick/ImageMagick/archive/7.1.0-33.tar.gz
+        sha256: dd18045614c19d11739311b207688c2cd480ed309d466685f1811f7a11df98ee
+        x-checker-data:
+          type: anitya
+          project-id: 1372
+          stable-only: true
+          url-template: https://github.com/ImageMagick/ImageMagick/archive/$version.tar.gz
+    cleanup:
+      - /bin
```
Comment (hfiguiere) [inline net.warsow.Warsow.yaml:33]: if you only need it at build time you can just add `*` as a cleanup rule.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: net.warsow.Warsow.yaml [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    build-options:
+      cflags: -fcommon
+    post-install:
+      - install -Dm0755 build/warsow.* ${FLATPAK_DEST}/bin/warsow
+      - install -Dm0755 build/wsw_server.* ${FLATPAK_DEST}/bin/warsow-server
+      - install -Dm0755 build/wswtv_server.* ${FLATPAK_DEST}/bin/warsow-tv-server
+      - mkdir -p ${FLATPAK_DEST}/share/warsow/libs
+      - cp build/libs/*.so ${FLATPAK_DEST}/share/warsow/libs
+      - mkdir -p ${FLATPAK_DEST}/share/icons/hicolor/256x256/apps/
+      - convert -strip ../icons/warsow256x256.xpm ${FLATPAK_DEST}/share/icons/hicolor/256x256/apps/warsow.png
```
Comment (hfiguiere) [inline net.warsow.Warsow.yaml:54]: I would prefer the .png in the repo and that it be submitted upstream rather that building ImageMagick just for that.

File: net.warsow.Warsow.yaml (hunk 3 above)
Comment (Mailaender) [reply net.warsow.Warsow.yaml:39]: It fails.

File: warsow.appdata.xml (hunk 2 above)
Comment (Mailaender) [reply warsow.appdata.xml:12]: Why? They are copyrighted as well!

**Comment by Mailaender** [comment]
bot, build net.warsow.Warsow

File: net.warsow.Warsow.yaml [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+      - type: archive
+        url: https://github.com/ImageMagick/ImageMagick/archive/7.1.0-33.tar.gz
+        sha256: dd18045614c19d11739311b207688c2cd480ed309d466685f1811f7a11df98ee
+        x-checker-data:
+          type: anitya
+          project-id: 1372
+          stable-only: true
+          url-template: https://github.com/ImageMagick/ImageMagick/archive/$version.tar.gz
+    cleanup:
+      - /*
```
Comment (hfiguiere) [inline net.warsow.Warsow.yaml:33]:
```suggestion
      - '*'
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: warsow.appdata.xml (hunk 2 above)
Comment (Mailaender) [reply warsow.appdata.xml:12]: Compare also https://flathub.org/apps/details/com.realm667.Wolfenstein_Blade_of_Agony

File: net.warsow.Warsow.yaml (hunk 5 above)
Comment (Mailaender) [reply net.warsow.Warsow.yaml:51]: Yes, it seems overkill.

File: flathub.json (hunk 1 above)
Comment (Mailaender) [reply flathub.json:2]: Let's give it a try. I think https://build.opensuse.org/package/live_build_log/games/warsow/openSUSE_Tumbleweed/aarch64 is failing for other reasons.

**Comment by Mailaender** [comment]
bot, build net.warsow.Warsow

File: flathub.json (hunk 1 above)
Comment (Mailaender) [reply flathub.json]:
```
../../source/as_callfunc_x64_gcc.cpp: In function ‘asQWORD X64_CallFunction(const asQWORD*, int, funcptr_t, asQWORD&, bool)’:
../../source/as_callfunc_x64_gcc.cpp:74:9: error: unknown register name ‘%rcx’ in ‘asm’
   74 |         __asm__ __volatile__ (
      |         ^~~~~~~
../../source/as_callfunc_x64_gcc.cpp:74:9: error: unknown register name ‘%rdx’ in ‘asm’
../../source/as_callfunc_x64_gcc.cpp:74:9: error: unknown register name ‘%rax’ in ‘asm’
../../source/as_callfunc_x64_gcc.cpp:74:9: error: unknown register name ‘%rsi’ in ‘asm’
../../source/as_callfunc_x64_gcc.cpp:74:9: error: unknown register name ‘%rdi’ in ‘asm’
../../source/as_callfunc_x64_gcc.cpp:74:9: error: unknown register name ‘%xmm7’ in ‘asm’
../../source/as_callfunc_x64_gcc.cpp:74:9: error: unknown register name ‘%xmm6’ in ‘asm’
../../source/as_callfunc_x64_gcc.cpp:74:9: error: unknown register name ‘%xmm5’ in ‘asm’
../../source/as_callfunc_x64_gcc.cpp:74:9: error: unknown register name ‘%xmm4’ in ‘asm’
../../source/as_callfunc_x64_gcc.cpp:74:9: error: unknown register name ‘%xmm3’ in ‘asm’
../../source/as_callfunc_x64_gcc.cpp:74:9: error: unknown register name ‘%xmm2’ in ‘asm’
../../source/as_callfunc_x64_gcc.cpp:74:9: error: unknown register name ‘%xmm1’ in ‘asm’
../../source/as_callfunc_x64_gcc.cpp:74:9: error: unknown register name ‘%xmm0’ in ‘asm’
make[3]: *** [makefile:125: obj/as_callfunc_x64_gcc.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/angelscript.dir/build.make:70: CMakeFiles/angelscript] Error 2
make[1]: *** [CMakeFiles/Makefile2:334: CMakeFiles/angelscript.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```

The builder also fails to abort and get stuck on arm:

> flatpak-builder: Cleanup warsowError: module warsow: Child process exited with code 2

**Comment by Mailaender** [comment]
bot, build net.warsow.Warsow

**Comment by barthalion** [comment]
I can't really see the linked Discord message. Can you make a screenshot?

File: net.warsow.Warsow.yaml [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+command: warsow
+rename-icon: warsow
+rename-desktop-file: warsow.desktop
+rename-appdata-file: warsow.appdata.xml
+finish-args:
+  - --socket=pulseaudio
+  - --socket=x11
```
Comment (barthalion) [inline net.warsow.Warsow.yaml:11]: If it uses SDL2, shouldn't it work fine with Wayland too?

File: flathub.json (hunk 1 above)
Comment (barthalion) [reply flathub.json:2]: Fedora builds it from source and has some patches available: https://src.fedoraproject.org/rpms/warsow/tree/rawhide

I tend to be practical about binary vs source builds, but I don't see the advantages for the binary in this case.

Comment (Mailaender) [reply flathub.json:2]: I also build it from source using the same tarball.

File: net.warsow.Warsow.yaml (hunk 7 above)
Comment (Mailaender) [reply net.warsow.Warsow.yaml]: I don't know. Every machine I own will crash when logging into a Wayland session.

**Comment by Mailaender** [comment]
No, I won't screenshot the chat dialog. However, as a result https://github.com/Warsow/qfusion/pull/53 was merged so maybe take that as proof.

**Comment by hfiguiere** [comment]
sadly with an incorrect app-id :-(

**Comment by hfiguiere** [comment]
since they were submitted they should have been submitted using the XDG convention that Flathub mandates.

(FWIW `rename-desktop` and friends is just a convenience for non conformant)

**Comment by Mailaender** [comment]
It was based upon my older https://build.opensuse.org/package/view_file/games/warsow/warsow.appdata.xml

**Comment by barthalion** [comment]
> No, I won't screenshot the chat dialog. However, as a result [Warsow/qfusion#53](https://github.com/Warsow/qfusion/pull/53) was merged so maybe take that as proof.

I ask, because we need to know if upstream is going to co-maintain it with you, and if yes, who should we invite to the repo.

**Comment by Mailaender** [comment]
They want to focus on their next big release and not help maintain this.

**Comment by barthalion** [comment]
bot, build net.warsow.Warsow

**Pull request closed without merging**